cql3/CqlParser.g4

```diff
@@ -582,19 +582,25 @@ assignmentElement
    ;
 
 assignmentSet
-   : syntaxBracketLc constant (syntaxComma constant)* syntaxBracketRc
+   : syntaxBracketLc ((constant (syntaxComma constant)*) |) syntaxBracketRc
    ;
 
 assignmentMap
    : syntaxBracketLc (constant syntaxColon constant) (constant syntaxColon constant)* syntaxBracketRc
    ;
 
 assignmentList
-   : syntaxBracketLs constant (syntaxColon constant)* syntaxBracketRs
+   : syntaxBracketLs constant (syntaxComma constant)* syntaxBracketRs
+   ;
+
+assignmentTuple
+   : syntaxBracketLr constant (syntaxComma constant)* syntaxBracketRr
+   | syntaxBracketLr constant (syntaxComma assignmentTuple)* syntaxBracketRr
+   | syntaxBracketLr assignmentTuple (syntaxComma assignmentTuple)* syntaxBracketRr
    ;
```

Comment:
I reccomed optimizind this by such a way:

```g4
assignmentTuple
   : syntaxBracketLr (
         constant ((syntaxComma constant)* | (syntaxComma assignmentTuple)*) |
         assignmentTuple (syntaxComma assignmentTuple)*
     ) syntaxBracketRr
   ;
```

Reply:
Updated by your advice, it's helpful.